ehrql/query_engines/in_memory_database.py

```diff
@@ -352,7 +352,7 @@ def filter(self, predicate):  # noqa A003
         patient_to_rows = {
             p: rows.filter(predicate[p]) for p, rows in self.patient_to_rows.items()
         }
-        return EventColumn({p: rows for p, rows in patient_to_rows.items() if rows})
+        return EventColumn({p: rows for p, rows in patient_to_rows.items()})
```

Comment:
This dict comprehension is redundant because it's just recreating the input dict. I think it would be clearer to write the whole function as a single line which then mirrors what we do elsewhere:
```suggestion
        return EventColumn(
            {p: rows.filter(predicate[p]) for p, rows in self.patient_to_rows.items()}
        )
```